datalad/distributed/ora_remote.py

```diff
@@ -510,6 +513,52 @@ def write_file(self, file_path, content, mode='w'):
             raise RIARemoteError("Could not write to {}".format(str(file_path)))
 
 
+class HTTPRemoteIO(object):
+    # !!!!
+    # This is not actually an IO class like SSHRemoteIO and LocalIO and needs
+    # respective RF'ing of special remote implementation eventually.
+    # We want ORA over HTTP, but with a server side CGI to talk to in order to
+    # reduce the number of requests. Implementing this as such an IO class would
+    # mean to have separate requests for all server side executions, which is
+    # what we do not want. As a consequence RIARemote class implementation needs
+    # to treat HTTP as a special case until refactoring to a design that fits
+    # both approaches.
+
+    # NOTE: For now read-only. Not sure yet whether an IO class is the right
+    # approach.
+
+    def __init__(self, ria_url, id):
+        assert ria_url.startswith("ria+http")
+        self.base_url = ria_url[4:]
+        if self.base_url[-1] == '/':
+            self.base_url = self.base_url[:-1]
+
+        self.base_url += "/" + id[:3] + '/' + id[3:]
+
+    def checkpresent(self, key_path):
+        # Note, that we need the path with hash dirs, since we don't have access
+        # to annexremote.dirhash from within IO classes
+
+        url = self.base_url + "/annex/objects/" + str(key_path)
+        response = requests.head(url)
+        return response.status_code == 200
+
+    def get(self, key_path, filename, progress_cb):
+        # Note, that we need the path with hash dirs, since we don't have access
+        # to annexremote.dirhash from within IO classes
+
+        url = self.base_url + "/annex/objects/" + str(key_path)
+        response = requests.get(url, stream=True)
+
+        with open(filename, 'wb') as dst_file:
+            bytes_received = 0
+            for chunk in response.iter_content(chunk_size=1024,
```

Comment:
How is this `chunk_size` value determined?

Reply:
Not at all. And the same is true for SSH. Wonder how to determine what's a reasonable size. And I guess we should make it configurable.

Reply:
Reading 1kb at a time seems very little. It wouldn't even fill up a single block on the file system.

Reply:
Absolutely agree. Any suggestion as to how to determine the size?

Reply:
ATM none other than to try a few and measure the impact, if there is any.

Reply:
Just tried to use `os.stat('.').st_blksize` to set the buffer to "preferred block size". 
Saw ~40MB/s w/o that change and ~140MB/s w/ that change. So - plausible and worthy starting point.

Reply:
Suspected potential there (and for SSH at the respective point), but a bit surprised by the effect size ...

Reply:
For me that size is 4KB. If I go to 16KB I got another factor 2. Suggests, that the effect actually has little to do with block size (which is kind of a relief, since my estimation, that it shouldn't do much for *that* reason wasn't that stupid after all ;-) ).

Reply:
Increasing further keeps getting faster, although the gain per additional buffer size slows down. Difference between 64KB and 80KB seems marginal. With 64KB I'm down to 4.9s for downloading 1.8GB in 32 files from local Apache, while it's been ~66s with current 1KB. (That's `datalad get` not `git-annex-get`)

Reply:
So, I'd suggest to make that a config (same for HTTP and SSH or separate?), defaulting to 64KB for now. With that config we can then have easy benchmarks for our setups, giving us a better idea. Agree @mih?

Reply:
My remark re blocksize wasn't meant to imply causation, just to say that it is very little. Making it configurable sound sensible. Providing protocol-specific tuning seems like overkill at this point.

Reply:
It seems rather than playing with the chunk size, we can leave it to the professionals: https://stackoverflow.com/a/39217788/1194609

Reply:
Tried that. In my local setup that is still 10% slower + leaves us with no progress reporting.